In [1]:
import sys
sys.path.append("..")
from utils.audio_utils import get_total_audio_duration,get_audio_info,get_duration_ffprobe
from pathlib import Path
from tqdm import tqdm
from utils.file_utils import save_json
from utils.transcription_utils import clean_and_verify_transcript_hi,get_transcription_info,get_stats,get_total_stats
from collections import Counter
from utils.conversions import sec_to_time
import csv

In [2]:
save_path=Path("/media/dataset-harddisk/munikumar/utils/audio_utils/data/fleurs/new")

In [3]:
dataset_dir=Path("/home/munikumar/tensorflow_datasets/downloads/extracted/TAR_GZ.xtreme_translations_FLEURS102_hi_inEaSxynH7BZG6ezARa-vP_4qmfH_hSq0E0ejV4mgFORM.tar.gz/hi_in")

In [4]:
def load_transcription(transcription_path):
    with open(transcription_path, "r", encoding="utf-8") as fp:
        reader=csv.reader(fp,delimiter='\t')
        
        transcription_data = {
            transcription[1]:transcription[3]
            for transcription in reader
        }
    return transcription_data

In [5]:
# train_transcription_data=load_transcription(dataset_dir/"train.tsv")
# get_transcription_info(train_transcription_data,dataset_dir/"audio"/"train",save_path,"train")
get_stats(save_path/"train_non_hindi_transcripts.json")

getting audio info about : new - train


  0%|          | 0/2120 [00:00<?, ?it/s]

100%|██████████| 2120/2120 [01:44<00:00, 20.31it/s]


**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 6:39:20
Loss duration : 2:33:57
Remaining duration : 4:5:22
Drop by percentage: 38.55

Total audio files : 2120
Files that would be drop: 748
Remaining audio files : 1372
Drop by percentage: 35.28
**************************************************
Non hindi tokes : ['!', '$', '%', "'", '+', ',', '-', '.', '/', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', ':', ';', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '£', '¥', '°', 'á', 'õ', 'ú', '।', '\u200d', '—', '”']


### Train Dataset
**************************************************
**Loss of audio data without transcription normalization:**
****************************************************
Total duration : 6:39:20  
Loss duration : 2:33:57  
Remaining duration : 4:5:22  
Drop by percentage: 38.55  

Total audio files : 2120  
Files that would be drop: 748  
Remaining audio files : 1372  
Drop by percentage: 35.28  
**************************************************
Non hindi tokes : ['!', '$', '%', "'", '+', ',', '-', '.', '/', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', ':', ';', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '£', '¥', '°', 'á', 'õ', 'ú', '।', '\u200d', '—', '”']


In [6]:
dev_transcription_data=load_transcription(dataset_dir/"dev.tsv")
get_transcription_info(dev_transcription_data,dataset_dir/"audio"/"dev",save_path,"dev")
get_stats(save_path/"dev_non_hindi_transcripts.json")

getting audio info about : new - dev


100%|██████████| 239/239 [00:12<00:00, 18.83it/s]


**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 0:42:45
Loss duration : 0:14:12
Remaining duration : 0:28:32
Drop by percentage: 33.25

Total audio files : 239
Files that would be drop: 63
Remaining audio files : 176
Drop by percentage: 26.36
**************************************************
Non hindi tokes : [',', '-', '.', '/', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'e', 'm', 'n', 'o', 'p', 'r', 's', 't', 'u', 'w', '।', '”']


### Dev Dataset
**************************************************
**Loss of audio data without transcription normalization:**
**************************************************
Total duration : 0:42:45  
Loss duration : 0:14:12  
Remaining duration : 0:28:32  
Drop by percentage: 33.25  

Total audio files : 239  
Files that would be drop: 63  
Remaining audio files : 176  
Drop by percentage: 26.36  
**************************************************
Non hindi tokes : [',', '-', '.', '/', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'e', 'm', 'n', 'o', 'p', 'r', 's', 't', 'u', 'w', '।', '”']

In [7]:
test_transcription_data=load_transcription(dataset_dir/"test.tsv")
get_transcription_info(test_transcription_data,dataset_dir/"audio"/"test",save_path,"test")
get_stats(save_path/"test_non_hindi_transcripts.json")

getting audio info about : new - test


100%|██████████| 418/418 [00:22<00:00, 18.37it/s]


**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 1:20:32
Loss duration : 0:27:39
Remaining duration : 0:52:52
Drop by percentage: 34.35

Total audio files : 418
Files that would be drop: 131
Remaining audio files : 287
Drop by percentage: 31.34
**************************************************
Non hindi tokes : ['!', '%', ',', '-', '.', '/', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', ':', ';', 'a', 'c', 'd', 'e', 'g', 'h', 'i', 'l', 'm', 'n', 'o', 'p', 's', 't', 'z', '।', '\u200d', '”']


### Test Dataset
**************************************************
**Loss of audio data without transcription normalization:**
**************************************************
Total duration : 1:20:32  
Loss duration : 0:27:39  
Remaining duration : 0:52:52  
Drop by percentage: 34.35  

Total audio files : 418  
Files that would be drop: 131  
Remaining audio files : 287  
Drop by percentage: 31.34  
**************************************************
Non hindi tokes : ['!', '%', ',', '-', '.', '/', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', ':', ';', 'a', 'c', 'd', 'e', 'g', 'h', 'i', 'l', 'm', 'n', 'o', 'p', 's', 't', 'z', '।', '\u200d', '”']


In [8]:
get_total_stats(save_path)

**************************************************
Loss of audio data for total dataset:
**************************************************
Total duration : 8:42:37
Loss duration : 3:15:50
Remaining duration : 5:26:47
Drop by percentage: 37.47

Total audio files : 2777
Files that would be drop: 942
Remaining audio files : 1835
Drop by percentage: 33.92


### Total Dataset
**************************************************
Loss of audio data for total dataset:
**************************************************
Total duration : 8:42:37  
Loss duration : 3:15:50  
Remaining duration : 5:26:47  
Drop by percentage: 37.47  

Total audio files : 2777  
Files that would be drop: 942  
Remaining audio files : 1835  
Drop by percentage: 33.92  